### PR TITLE
DataGrid - Fix cell validation in Row mode (T872003)

### PR DESCRIPTION
**Pull request by NickMitrokhin**
1. EditingController: wrapped showHighlighting into an extra method - highlightDataCell.
2. Added a test for this case.

### CODE REVIEW

File: js/ui/grid_core/ui.grid_core.validating.js [hunk 1]
```diff
@@ -798,6 +798,22 @@ module.exports = {
...
+                highlightDataCell: function($cell, parameters) {
+                    const isEditableCell = parameters.setValue;
+                    const cellModified = this.isCellModified(parameters);
+
+                    if(!cellModified && isEditableCell) {
+                        const result = this.getController('validating').getCellValidationResult({
+                            rowKey: parameters.key,
+                            columnIndex: parameters.column.index
+                        });
+                        const skipValidation = parameters.row.isNewRow || !isDefined(result);
```
Comment (vconst) [inline js/ui/grid_core/ui.grid_core.validating.js]: Extract `isDefined(result)` to variable `isValidated`

File: js/ui/grid_core/ui.grid_core.validating.js [hunk 2]
```diff
@@ -798,6 +798,22 @@ module.exports = {
                     }
                 },
 
+                highlightDataCell: function($cell, parameters) {
+                    const isEditableCell = parameters.setValue;
+                    const cellModified = this.isCellModified(parameters);
+
+                    if(!cellModified && isEditableCell) {
+                        const result = this.getController('validating').getCellValidationResult({
```
Comment (vconst) [inline js/ui/grid_core/ui.grid_core.validating.js]: result -> validationResult

**Review by vconst — CHANGES_REQUESTED** [review]

File: js/ui/grid_core/ui.grid_core.validating.js (hunk 1 above)
Comment (NickMitrokhin) [reply js/ui/grid_core/ui.grid_core.validating.js]: OK

File: js/ui/grid_core/ui.grid_core.validating.js (hunk 2 above)
Comment (NickMitrokhin) [reply js/ui/grid_core/ui.grid_core.validating.js]: OK

**Review by vconst — APPROVED** [review]

**Pull request merged**